### ISSUE TITLE
subset() by expression of any gene in list

### ISSUE THREAD

**Issue by ZacharyDeBruine** [satijalab/seurat#2327]
I'm very new to Seurat and R, but worked through all tutorials and am impressed by the package. 

I need to subset a Seurat object to contain only cells that express any of several genes of interest (not all of them, but any of them).

I'm using Mouse Cell Atlas (mca) data as described [here](https://satijalab.org/seurat/v3.1/mca.html). I want to subset the object (`mca`) based on expression of at least one of the genes in an array (`genes`). I have tried:

```
#create Seurat object from sparse R matrix data
mca.matrix <- readRDS(file = "MCA_merged_mat.rds")
mca.metadata <- read.csv(file = "MCA_All-batch-removed-assignments.csv", row.names = 1)
mca <- CreateSeuratObject(counts = mca.matrix, meta.data = mca.metadata, project = "MCA")
mca <- NormalizeData (mca, normalization.method = "LogNormalize", scale.factor = 10000)

#Remove all cells that are not annotated
mca <- subset(mca, cells = names(which(!is.na(mca$ClusterID))))

#Array of genes of interest
genes <- c("Apc","Wnt3a","FZD5")

#Generate separate subsets for each gene of interest
subset1 <- SubsetData(mca, subset.name = genes[0], low.threshold = 1)
subset2 <- SubsetData(mca, subset.name = genes[1], low.threshold = 1)
subset3 <- SubsetData(mca, subset.name = genes[2], low.threshold = 1)

#Now merge these three Seurat objects together again
merged_subset12 <- merge(subset1, y = subset2)
merged_subsets <- merge(subset12, y = subset3)
```

This is not an acceptable solution for me because many cells co-express Apc, Wnt3a, and FZD5 and therefore `merged_subsets` contains duplicate cells with automatically renamed values. **How can cells containing any of the genes of interest--but not necessarily all of them--be subsetted into a new Seurat object?**

My ultimate goal is to export the resulting Seurat object to an R table containing only the genes of interest as columns and cells expressing any of those genes as rows, with normalized read data. If anyone could point out how to do that it would be so much appreciated!

I have seen [this](https://bioinformatics.stackexchange.com/questions/4232/subset-on-multiple-genes-in-seurat) question, but it was not helpful.

**Comment by ZacharyDeBruine** [comment]
I was able to resolve my issue subsetting data by pinpointing what was wrong with the answer in [this](https://bioinformatics.stackexchange.com/questions/4232/subset-on-multiple-genes-in-seurat) post.

I found that the strategy of creating an array of `cell_names` after running `SubsetData` on individual genes was working, but that I needed to use `SubsetData(mca, cells = cell_names)` instead of `SubsetData(mca, cells.use = cell_names)`. This is apparently a Seurat V3 vs V2 thing.

I still would greatly appreciate guidance on exporting the resulting Seurat object to an R table containing only the genes of interest as columns and cells expressing any of those genes as rows, with normalized read data. Thanks!
